src/rust/bitbox02/src/ui/types.rs

```diff
@@ -99,9 +101,30 @@ impl<'a> ConfirmParams<'a> {
     }
 }
 
+/// Helper struct to convert from a list of Rust strings to a list of C strings.
+pub struct CWords {
+    /// Buffers to hold the C strings, including null terminators.
+    buffers: Vec<Vec<u8>>,
+    /// List of pointers to the C strings, pointing into `buffers`.
+    words: Vec<*const util::c_types::c_char>,
+}
+
+impl CWords {
```

Comment:
Would be very nice to have some CWords tests.

Reply:
Done in a new commit (and moved to util.rs)